app/controllers/api/event_dates_controller.rb

```diff
@@ -7,10 +7,14 @@ class EventDatesController < Api::BaseController
 
     # GET /api/event_dates/:id
     def show
-      render json:
-        ActiveModelSerializers::SerializableResource.new(
-          @event_date, event_hours: true, event_slots: true
-        ).as_json
+      if @event_date.valid?
+        render json:
+          ActiveModelSerializers::SerializableResource.new(
+            @event_date, event_hours: true, event_slots: true
+          ).as_json
+      else
+        render json: { errors: @event_date.errors }
+      end
     end
 
```

Comment:
For discussion:

@event_date needs to be unscoped for the scope validations to work.  Two possible methods and two variations of the second method.  First is unscope-validate-scope in before action.

```
def set_event_date
      # handle scope related errors by unscoping
      @event_date = EventDate.unscoped.find(params[:event_date_id])
      if @event_date.valid?
        # if no errors perform scoped find
        @event_date = EventDate.find(params[:event_date_id])
      else
        render json: { errors: @event_date.errors }
      end
  end
```

Second is unscope in before action and leave unscoped.  This requires adding all validation checks for all scopes, instead of just the two we are doing now.  In this case the validation could be done in the before action or in the show method (below shows in the before action). 

```
def set_event_date
      # handle scope related errors by unscoping
      @event_date = EventDate.unscoped.find(params[:event_date_id])
      if @event_date.valid? == false
        render json: { errors: @event_date.errors }
      end
  end
```

A third method (based on the second method) would be to throw exceptions in the scope related exceptions by running the individual scopes instead of duplicating the scope logic in the validations which we have already seen is error prone and requires changing validations if scopes change.  This would mean that the first scope that fails would prevent additional errors from being shown but the scopes could be ordered by probability of occurring, e.g., published_end_datetime, status_id, etc.